src/lambdas/breeds/breeds-get.test.ts

```diff
@@ -0,0 +1,93 @@
+import fetch, { FetchError } from 'node-fetch'
+import { handler, DogBreedsListResponse } from './breeds-get'
+
+const mockedFetch: jest.Mock = fetch as any
+
+jest.mock('node-fetch')
+
+describe('breeds-get handler', () => {
+  afterEach(() => {
+    mockedFetch.mockClear()
+  })
+
+  describe('when the api returns the expected value', () => {
+    const mockPayload = {
+      message: {
+        sheepdog: ['english', 'shetland'],
+        beagle: [],
+      },
+    }
+    beforeEach(() => {
+      mockedFetch.mockReturnValueOnce({
+        ok: true,
+        json: () => {
+          return mockPayload
+        },
+      })
+    })
+
+    it('returns payload from fetch request', async () => {
+      const response = (await handler()) as DogBreedsListResponse
+      expect(response.body).toEqual(
+        expect.arrayContaining(['english sheepdog', 'shetland sheepdog', 'beagle']),
+      )
+      expect(response.body).toHaveLength(3)
+    })
+  })
+
+  describe('when the api returns an error response', () => {
+    beforeEach(() => {
+      mockedFetch.mockReturnValueOnce({
+        ok: false,
+        status: 408,
+        statusText: 'Request Timeout',
+        json: () => {
+          return null
+        },
+      })
```

Comment:
Are we sure this is what `fetch` returns on timeout? See my note on abort/race below

Reply:
This is not a fetch error simulation, but a server response that I simulated. I switch this to a 404 to make it more obvious.